File: PetParameters.cs
using System;
using System.Collections.Generic;

public class PetParameters
{
    private int hunger, fatigue, health, mood;

    public int Hunger => hunger;
    public int Fatigue => fatigue;
    public int Health => health;
    public int Mood => mood;

    public PetParameters(int hunger = 50, int fatigue = 50, int health = 100, int mood = 50)
    {
        SetHunger(hunger);
        SetFatigue(fatigue);
        SetHealth(health);
        SetMood(mood);
    }

    public void UpdateOverTime()
    {
        hunger += 10;
        fatigue += 5;
        if (hunger > 75) health -= 5;
        if (fatigue > 70) mood -= 10;
    }

    public bool CheckCriticalLevels() => health <= 20;

    public Dictionary<string, int> GetParameters() =>
        new() { ["hunger"] = hunger, ["fatigue"] = fatigue, ["health"] = health, ["mood"] = mood };

    public void SetHunger(int value) => hunger = Math.Clamp(value, 0, 100);
    public void SetFatigue(int value) => fatigue = Math.Clamp(value, 0, 100);
    public void SetHealth(int value) => health = Math.Clamp(value, 0, 100);
    public void SetMood(int value) => mood = Math.Clamp(value, 0, 100);
}

File: PetParameters.cpp
#include "PetParameters.h"

void PetParameters::updateOverTime() {
    hunger += 10;
    fatigue += 5;
    if (hunger > 75) health -= 5;
    if (fatigue > 70) mood -= 10;
}

bool PetParameters::checkCriticalLevels() const {
    return health <= 20;
}

std::map<std::string, int> PetParameters::getParameters() const {
    return { {"hunger", hunger}, {"fatigue", fatigue}, {"health", health}, {"mood", mood} };
}

int PetParameters::getHunger() const { return hunger; }
int PetParameters::getFatigue() const { return fatigue; }
int PetParameters::getHealth() const { return health; }
int PetParameters::getMood() const { return mood; }

void PetParameters::setHunger(int value) {
    hunger = (value < 0) ? 0 : (value > 100 ? 100 : value);
}
void PetParameters::setFatigue(int value) {
    fatigue = (value < 0) ? 0 : (value > 100 ? 100 : value);
}
void PetParameters::setHealth(int value) {
    health = (value < 0) ? 0 : (value > 100 ? 100 : value);
}
void PetParameters::setMood(int value) {
    mood = (value < 0) ? 0 : (value > 100 ? 100 : value);
}